src/export/InstanceExporter.ts

```diff
@@ -281,6 +281,26 @@ export class InstanceExporter implements Fishable {
       fshDefinition
     );
     cleanResource(instanceDef);
+
+    // check for another instance of the same type with the same id
+    // see https://www.hl7.org/fhir/resource.html#id
+    // instanceDef has already been added to the package, so it's fine if it matches itself
+    // inline instances are not written to their own files, so those can be skipped in this
+    if (
+      this.pkg.instances.some(
+        instance =>
+          instance._instanceMeta.usage !== 'Inline' &&
```

Comment:
I'm wondering about one more case:
```
Instance: MyPatient1
InstanceOf: Patient
* id = "testing"
* name.family = "hello"

Instance: MyPatient2
InstanceOf: Patient
Usage: #inline
* id = "testing"
* name.family = "hello there"
```
In this case, `MyPatient2` gets flagged for having the same id as `MyPatient1`. I *think* that this should not be the case right? Since `MyPatient2` would not be existing as its own entity. You'd have to be a pretty cruel user to go around doing this and then getting mad that it doesn't work, so sorry for this edge case on an edge case. But what do you think? Should this case still be an error or not?

Reply:
This case should not be an error, since there will only be one resource on the server with that id, meaning that the FHIR rules about id uniqueness is satisfied. And importantly, it currently does not throw an error when the instances are in the opposite order. I'll handle this case and add a test.